tests/Doctrine/Tests/ORM/EntityManagerTest.php

```diff
@@ -191,12 +191,13 @@ public function testTransactionalAcceptsVariousCallables()
         self::assertSame('callback', $this->_em->transactional(array($this, 'transactionalCallback')));
     }
 
-    public function testTransactionalThrowsInvalidArgumentExceptionIfNonCallablePassed()
+    public function testTransactionalAcceptsReturnFalse()
     {
-        $this->expectException(\InvalidArgumentException::class);
-        $this->expectExceptionMessage('Expected argument of type "callable", got "object"');
+        $return = $this->_em->transactional(function ($em) {
+            return false;
+        });
 
-        $this->_em->transactional($this);
+        self::assertEquals(false, $return);
```

Comment:
@Ocramius As this would add third test for the same thing I refactored them to a dataProvider.